app/src/main/java/com/soen490chrysalis/papilio/viewModel/UserProfileViewModel.kt

```diff
@@ -0,0 +1,133 @@
+package com.soen490chrysalis.papilio.viewModel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.auth.EmailAuthProvider
+import com.google.firebase.auth.FirebaseAuth
+import com.soen490chrysalis.papilio.repository.users.IUserRepository
+import kotlinx.coroutines.launch
+import java.util.regex.Pattern
+
+class UserProfileViewModel(private val userRepository : IUserRepository) : ViewModel()
+{
+    var userObject : MutableLiveData<GetUserResponse> = MutableLiveData<GetUserResponse>()
+    var passwordChangeResult : MutableLiveData<String> = MutableLiveData<String>()
+    private var editedFields : MutableMap<String, kotlin.Any> = mutableMapOf<String, kotlin.Any>()
+
+    fun getUserByFirebaseId()
+    {
+        viewModelScope.launch {
+            val userResponse = userRepository.getUserByFirebaseId()
+            userObject.value = GetUserResponse(true, userResponse?.user)
+        }
+    }
+
+    fun addEditedField(fieldName: String, fieldValue: kotlin.Any)
+    {
+        editedFields.put(fieldName, fieldValue);
+    }
+
+    fun isEditedFieldsEmpty() : Boolean
+    {
+        return editedFields.isEmpty()
+    }
+
+    fun updateUserProfile()
+    {
+        if(editedFields.isEmpty()) return
+
+        viewModelScope.launch {
+            val userResponse = userRepository.updateUser(editedFields)
+
+            val afterUpdateResponse = userRepository.getUserByFirebaseId()
+
+            userObject.value = GetUserResponse(true, afterUpdateResponse?.user)
+
+            editedFields.clear()
+        }
+    }
+
+    fun validatePhoneNumber(number : String): String?
+    {
+        if(number.length < 10)
+        {
+            return "Please enter a valid phone number (10 digits)"
+        }
+
+        return null
+    }
+
+    fun validateCountryCode(number : String): String?
+    {
+        if(number.length > 5 || number.isEmpty())
+        {
+            return "Please enter a valid country code (without the plus sign)"
+        }
+
+        return null
+    }
+
+    fun validatePassword(password : String) : String?
```

Comment:
Maybe we should put these validation functions in their own file rather than inside view models since they keep getting reused